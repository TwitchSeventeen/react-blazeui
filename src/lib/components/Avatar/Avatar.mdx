---
name: Avatar
route: /components/avatar
menu: Components
---

import { Playground, PropsTable } from 'docz'
import Avatar from './index';

# Avatar

Avatars are useful for chat UIs, profile pictures and account photos.

## Basic usage
Use the **src** and **alt** prop to define the image to show. The **size** prop sets the size, medium is the default.
<Playground>
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" size="super" />
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" size="xlarge" />
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" size="large" />
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" />
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" size="small" />
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" size="xsmall" />
</Playground>

## 2 Images
You can add a smaller secondary image with the **src2** and **alt2** props.
<Playground>
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" alt2="placeholder" src2="http://placehold.it/1024x963/BADA55/FFF" size="super" />
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" alt2="placeholder" src2="http://placehold.it/1024x963/BADA55/FFF" size="xlarge" />
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" alt2="placeholder" src2="http://placehold.it/1024x963/BADA55/FFF" size="large" />
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" alt2="placeholder" src2="http://placehold.it/1024x963/BADA55/FFF" />
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" alt2="placeholder" src2="http://placehold.it/1024x963/BADA55/FFF" size="small" />
    <Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" alt2="placeholder" src2="http://placehold.it/1024x963/BADA55/FFF" size="xsmall" />
</Playground>


## Initials
Instead of showing images, you can show text using the **initials** prop.
<Playground>
    <Avatar initials="LO" size="super" />
    <Avatar initials="LO" size="xlarge" />
    <Avatar initials="LO" size="large" />
    <Avatar initials="LO" />
    <Avatar initials="LO" size="small" />
    <Avatar initials="LO" size="xsmall" />
</Playground>

## Properties
<PropsTable of={Avatar} />