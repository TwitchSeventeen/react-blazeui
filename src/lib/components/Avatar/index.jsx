import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Image from '../Image';

function Avatar(props) {
  const {
    initials, className, size, src, alt, src2, alt2, ...other
  } = props;
  const classes = classNames(
    'c-avatar',
    `u-${size}`,
    className,
  );
  return (
    <div className={classes} data-text={initials} {...other}>
      { src
          && <Image className="c-avatar__img" src={src} alt={alt} />
        }

      { src2
        && <Image className="c-avatar__img" src={src2} alt={alt2} />
        }
    </div>
  );
}

Avatar.propTypes = {
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * The avatar size
   */
  size: PropTypes.oneOf([
    'xsmall',
    'small',
    'medium',
    'large',
    'xlarge',
    'super',
  ]),
  /**
   * Text to display in the middle of the circle
   */
  initials: PropTypes.string,
  /**
   * URL to the image to display
   */
  src: PropTypes.string,
  /**
   * Alternate text for the image
   */
  alt: PropTypes.string,
  /**
   * URL for the secondary image to display
   */
  src2: PropTypes.string,
  /**
   * Alternate text for the secondary image
   */
  alt2: PropTypes.string,
};

Avatar.defaultProps = {
  initials: null,
  className: null,
  size: null,
  src: null,
  alt: null,
  src2: null,
  alt2: null,
};

export default Avatar;
