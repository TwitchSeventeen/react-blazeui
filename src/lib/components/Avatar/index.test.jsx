import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Avatar from './index';

describe('Avatar', () => {
  it('should be defined', () => {
    expect(Avatar).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" size="super" />);

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with two images', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Avatar alt="placeholder" src="http://placehold.it/1024x963/3B6DC6/FFF" alt2="placeholder" src2="http://placehold.it/1024x963/BADA55/FFF" size="super" />);

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with initials', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Avatar initials="LKO" />);

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
