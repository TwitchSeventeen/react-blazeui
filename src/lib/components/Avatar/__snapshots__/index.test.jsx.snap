// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<div
  className="c-avatar u-super"
  data-text={null}
>
  <Image
    alt="placeholder"
    className="c-avatar__img"
    filter={null}
    height={null}
    likes={false}
    src="http://placehold.it/1024x963/3B6DC6/FFF"
    user={null}
    width={null}
  />
</div>
`;

exports[`Avatar renders correctly with initials 1`] = `
<div
  className="c-avatar u-null"
  data-text="LKO"
/>
`;

exports[`Avatar renders correctly with two images 1`] = `
<div
  className="c-avatar u-super"
  data-text={null}
>
  <Image
    alt="placeholder"
    className="c-avatar__img"
    filter={null}
    height={null}
    likes={false}
    src="http://placehold.it/1024x963/3B6DC6/FFF"
    user={null}
    width={null}
  />
  <Image
    alt="placeholder"
    className="c-avatar__img"
    filter={null}
    height={null}
    likes={false}
    src="http://placehold.it/1024x963/BADA55/FFF"
    user={null}
    width={null}
  />
</div>
`;
